### PR TITLE
Allow to disable TORCH_COMPILE_DEBUG and TORCHDYNAMO_SUPPRESS_ERRORS by setting them to 0

### PR DESCRIPTION
**Pull request by eeggl**
Summary:
Converting "0" to bool results in True, change it to explicitly
check for "0" and "1" values to be consistent with the rest of the codebase.

```
$ export TORCHDYNAMO_SUPPRESS_ERRORS=1; python3 -c 'import os; print(bool(os.environ.get("TORCHDYNAMO_SUPPRESS_ERRORS", False)))'
True
$ export TORCHDYNAMO_SUPPRESS_ERRORS=0; python3 -c 'import os; print(bool(os.environ.get("TORCHDYNAMO_SUPPRESS_ERRORS", False)))'
True
```

Differential Revision: D42501779

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @anijain2305 @mlazos @soumith @yanboliang @chunyuan-w @desertfire

### CODE REVIEW

**Review by ngimel — APPROVED** [review]

**Comment by eeggl** [comment]
@ngimel I think I need an approval to start workflows before we can land it.

**Comment by eeggl** [comment]
Rebased the PR

**Review by ngimel — APPROVED** [review]

**Pull request closed without merging**
